packages/grant-explorer/src/features/api/round.ts

```diff
@@ -67,6 +67,77 @@ export type RoundProject = {
   payoutAddress: string;
 };
 
+export type ProjectsWithRoundData = {
+  roundMeta: RoundMetadata;
+  roundEndTime: Date;
+  projectsMeta: Project[];
+};
+
+export async function getProjects(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  chainId: any,
+  roundId: string,
+  projectId: number[],
+  withRoundMeta?: boolean
+): Promise<Project[] | ProjectsWithRoundData> {
+  const res = await graphql_fetch(
+    `
+      query GetProjectByIndex($roundId: String, $projectId: [Int]) {
+        round(id: $roundId) {
+          roundMetaPtr {
+            pointer
+          }
+          roundEndTime
+          projects(where: {applicationIndex_in: $projectId}) {
+            id
+            project
+            status
+            applicationIndex
+            metaPtr {
+              protocol
+              pointer
+            }
+          }
+        }
+      }
+    `,
+    chainId,
+    { roundId, projectId }
+  );
+
+  const projectsRes = res.data.round.projects as RoundProjectResult[];
+
+  const approvedProjects = projectsRes.map((project) => {
+    return {
+      ...project,
+      status: convertStatus(project.status),
+    };
+  });
+
+  const fetchApprovedProjectMetadata: Promise<Project>[] = approvedProjects.map(
+    (project: RoundProjectResult) =>
+      fetchMetadataAndMapProject(project, chainId)
+  );
+
+  const projectsMetadata = await Promise.all(fetchApprovedProjectMetadata);
```

Comment:
Use SWR for proper state handling and loading states.

Also, if we fetch the metadata in a separate hook we can fetch them in the component when it's rendered. This will improve loading times.